src/core/Akka/Actor/Props.cs

```diff
@@ -554,20 +538,29 @@ public virtual ActorBase NewActor()
         ///     Creates a copy of the current instance.
         /// </summary>
         /// <returns>The newly created <see cref="Akka.Actor.Props" /></returns>
-        protected virtual Props Copy()
+        private Props Copy()
         {
-            return new Props(_producer, Deploy, Arguments) { SupervisorStrategy = SupervisorStrategy };
+            return new Props(_producer, Deploy, Type, Arguments) { SupervisorStrategy = SupervisorStrategy };
         }
 
         [Obsolete("we should not be calling this method. Pass in an explicit IIndirectActorProducer reference instead.")]
-        private static IIndirectActorProducer CreateProducer(Type type, object[] args)
+        private static (IIndirectActorProducer, Type, object[] args) CreateProducer(Type type, object[] args)
         {
-            if (type == null) return DefaultProducer.Instance;
+            if (type is null)
+                return (InvalidProducer.Default, typeof(EmptyActor), NoArgs);
 
             if (typeof(IIndirectActorProducer).IsAssignableFrom(type))
-                return Activator.CreateInstance(type, args).AsInstanceOf<IIndirectActorProducer>();
+            {
+                var producer = (IIndirectActorProducer)Activator.CreateInstance(type, args);
+                if (producer is IIndirectActorProducerWithActorType p)
+                    return (producer, p.ActorType, NoArgs);
+
+                //maybe throw new ArgumentException($"Unsupported legacy actor producer [{type.FullName}]", nameof(type));
```

Comment:
here I am not sure, maybe to throw is better

Reply:
If it's _actually_ unsupported now, we should throw here.